src/libstd/net/udp.rs

```diff
@@ -180,6 +180,23 @@ impl UdpSocket {
         }
     }
 
+    /// Returns the socket address of the remote peer this socket was connected to.
```

Comment:
What happened if the UDP socket isn't connected (UDP can work without a connection)?

Reply:
It will return the error `ENOTCONN`, would it be better to have it return a `Result<Option<SocketAddr>>`?

Reply:
@LinusU It's fine returning `Result<SocketAddr>` as long as the behavior when not connected is properly documented.

Reply:
Added some more documentation, what do you think? ☺️